### ISSUE TITLE
Debug sidebar - Filter this node - doesn't...

### ISSUE THREAD

**Issue by dceejay** [node-red/node-red#3218]
### Current Behavior

Two related issues
1) If I have the debug sidebar open with several nodes pointed at it...  (in all nodes or current flow mode) - and then use the select button on right of a debug message 
![image](https://user-images.githubusercontent.com/5375409/138594709-bdacf060-297b-40fc-b445-9ccf677c8a53.png)
to select "Filter this node" - It does not... as the main select 
![image](https://user-images.githubusercontent.com/5375409/138594733-0717ab5d-2ae9-4911-a3fd-5bba2468d9fb.png)
 stays as it was.
2) The main select is set as "selected nodes" and you then select "filter this node" - it starts to filter - but only using the first node it finds in the selected nodes option menu - again not the one you actually select.

(It would also be nice if the "selected node" selection sub-menu allowed selection of the nodes by clicking in the workspace (like the catch and status nodes do) - to make it easier to select between debug nodes that are all called msg.payload by default :-)...

### Expected Behavior

In case 1) I would expect the selection of "Filter this node" to force the overall filter to be selected nodes only mode
and then for case 2) I would expect it to then correctly select the node chosen to filter on.

### Steps To Reproduce

For case 1) leave debug in all nodes mode - wait for debug to appear - select Filter this node option for either... note both messages still appear... 
For case 2) set selected nodes mode - wait for debug - select Filter this node on the "From timestamp" debug node - see that the debug only contains messages "From hello" instead.... (and can be seen incorrectly selected in the selected nodes top dropdown.)

### Example flow

```
[{"id":"f99899a2e1cc0e08","type":"inject","z":"4aa2ed2fd1b11362","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"8","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"Hello","payloadType":"str","x":160,"y":100,"wires":[["f7f67fbef3c0d07b"]]},{"id":"f7f67fbef3c0d07b","type":"debug","z":"4aa2ed2fd1b11362","name":"From hello node","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":530,"y":100,"wires":[]},{"id":"dff6ecb128518c17","type":"inject","z":"4aa2ed2fd1b11362","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"9","crontab":"","once":false,"onceDelay":0.1,"topic":"","payloadType":"date","x":180,"y":200,"wires":[["c1d12b646a328684"]]},{"id":"c1d12b646a328684","type":"debug","z":"4aa2ed2fd1b11362","name":"From timestamp node","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":540,"y":200,"wires":[]}]
```

### Environment

- Node-RED version: 2.1.1
- Node.js version: 14.18.1
- npm version: 6.14.15
- Platform/OS: Mac OS
- Browser: Chrome

**Comment by tablatronix** [comment]
I have no idea what this feature is supposed to do, it appears to do nothing..

**Comment by knolleary** [comment]
Fixed since NR 3.1.1
